WordPress/Classes/Utility/Migration/ContentMigrationCoordinator.swift

```diff
@@ -157,6 +164,27 @@ private extension ContentMigrationCoordinator {
             self.cleanupExportedDataIfNeeded()
         }
     }
+
+    /// A "middleware" logic that attempts to record (or clear) any migration error to the App Group space
+    /// before calling the completion block.
+    ///
+    /// - Parameters:
+    ///   - result: The `Result` object from the export process.
+    ///   - completion: Closure that'll be executed after the process completes.
+    func processResult(_ result: Result<Void, ContentMigrationCoordinatorError>, completion: ((Result<Void, ContentMigrationCoordinatorError>) -> Void)?) {
+        switch result {
+        case .success:
+            sharedPersistentRepository?.removeObject(forKey: .exportErrorSharedKey)
+
+        case .failure(let error):
+            guard let description = error.errorDescription else {
+                break
+            }
+            sharedPersistentRepository?.set(description, forKey: .exportErrorSharedKey)
```

Comment:
I think if we want to present an Alert ( or Error Screen ) on the Jetpack side, we should store the `rawValue` of `ContentMigrationCoordinatorError` instead of `errorDescription`, then we let the UI side to decide how to format the error and display it to the user. What do you think?

Reply:
Good point. This is now addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/19787/commits/95a1ccc320af53cbc34a2689d68fcf1868b4858b!